sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs

```diff
@@ -819,7 +823,7 @@ private static void ThrowSerializationFailed(string propertyName, KeyValuePair<s
         {
             if (symbol.Equals(AmqpMessageConstants.Uri))
             {
-                return new Uri((string)value);
+                return new Uri((string)value, UriKind.RelativeOrAbsolute);
```

Comment:
I assume uri.AbsoluteUri still works as long as it is actually Absolute?

Reply:
Yup.  That gets handled internally by `Uri` at creation time.